### ISSUE TITLE
_intPin1 and _intPin2

### ISSUE THREAD

**Issue by vvirag** [kriswiner/BMA280#2]
The BMA280 class has a _intPin1 and _intPin2 member, which are set in the constructor.
However, I don't see them being read out afterwards. What are they used for then?

(I'm trying to port this code to another platform, and I'm not sure how to set up the interrupts to set the newData and newTap flags -- since I don't understand which pins should these interrupts coupled to.)

Thanks for any help!

**Comment by kriswiner** [comment]
These serve no function in this implementation and can be removed.

**Comment by vvirag** [comment]
Yes, that's what I thought, thanks :)

I'm still puzzled though where the 4 and 1 come from regarding these lines:
`#define intPin1 4
#define intPin2 1`

-- since later the `newData` and `newTap` flags are getting set based on the interrupts when these 2 pins are rising.

**Comment by kriswiner** [comment]
These are the GPIOs I am using for these BMA280 interrupts.

**Comment by vvirag** [comment]
Ah, wait, you're saying these are external GPIOs on the Arduino that you use when you actually want to read out (from the application code) the tap/move states of the accelerometer?

**Comment by kriswiner** [comment]
Yes, I am using the interrupts as input to the STM32L4 MCU for data ready
and tap detection (or other functions). I am just saying _intPin1 and
_intPin1 serve no purpose in the .cpp file or functions since the
interrupts are managed in the main sketch.

**Comment by vvirag** [comment]
I'm sorry maybe I still don't get it right. This means you are polling accelerometer data from the MCU, right? Meaning that something - that is unrelated to the accelerometer - will pull up the pin1 (=1 on the MCU) and pin2 (=4 on the MCU) whenever you decide you want to read out the accelerometer status.

Or I'm completely wrong..? :)

**Comment by kriswiner** [comment]
Completely wrong.

The BMA280 interrupts are configured to pulse whenever data is ready or tap
is detected. This lets the MCU know it is time to read the data, handle the
taps, etc.

Interrupts 101? Look at Arduino.cc for a tutorial....

**Comment by vvirag** [comment]
I have some clue about interrupts, and I'm not even working on Arduino. And yes, it seemed strange, but I just still don't understand how exactly are the 
> BMA280 interrupts configured.

As far as I can tell that configuration is happening in the `BMA280::initBMA280` function. But there are lots of magic (hexa)numbers there, and even though the BMA280 documentation is in front of me, I haven't been able to completely understand it.

And this goes back to my original question on where the 1 and 4 are coming from (as defined in the .ino). And how those values are reflected in that `initBMA280` method.

**Comment by kriswiner** [comment]
Last time.

Interrupts are outputs from the BMA280 (must be configured according to the
BMA280 data sheet) specified in initBMA280

and are inputs to the MCU, which GPIO pins (in this case pin 1 and pin 4)
must be specified and configured in the main sketch.

This is true whether you are using Arduino or anything else.

**Comment by vvirag** [comment]
I appreciate your answers, really. And I agree with everything you just said, that's all clear.

I guess I just don't see where (which line in your code) the MCU GPIO1 and MCU GPIO4 gets connected to the INT1 and INT2 interrupts of the BMA280.

Is my question more clear this way?

**Comment by vvirag** [comment]
Oh my, it just hit me. This connection-configuration that I was looking for in the code.. It's not in the code, right? You just have them physically connected, right? Ah.. Sry for the troubles.

**Comment by cleonb32** [comment]
looking at page 50 of the data sheet, i don't think this "destination[0] = ((int16_t)rawData[1] << 8) | rawData[0];" is correct. that will give you a 16 bit number (should be 14 bits).; the lower 2 bits of the LSB are as follow: bit 1- undefined, bit 0: new_data flag. should it be something like: destination[0] = ((int16_t)(rawData[1] << 6) | (rawData[0]>>2));" ?

**Comment by kriswiner** [comment]
In the main loop we divide by 4:

     BMA280.readBMA280AccelData(accelCount); // get 14-bit signed accel data

     // Now we'll calculate the accleration value into actual g's
     ax = (float)accelCount[0]*aRes/4.0f;  // get actual g value, this
depends on scale being set
     ay = (float)accelCount[1]*aRes/4.0f;
     az = (float)accelCount[2]*aRes/4.0f;

this is the same as >> 2. So either way you want to do this is fine.

**Comment by cleonb32** [comment]
thanks.  looking at the datasheets I could not find your reference to 8192 (0x2000).

**Comment by kriswiner** [comment]
" I could not find your reference to 8192 (0x2000)."

??Huh??

**Comment by cleonb32** [comment]
"_aRes = 2.0f/8192.0f;// per data sheet" i could not find 8192.0f in the datasheet

**Comment by kriswiner** [comment]
14-bit signed integer means maximum value of the data is 2^13.

**Comment by cleonb32** [comment]
thanks
